packages/core/cache/nuxt/index.js

```diff
@@ -1,93 +1,132 @@
-  }
+/**
+ * Creates custom renderer to use cache as much as possible.
+ */
+function createRenderer (renderFn) {
+  const renderer = this.nuxt.renderer;
+  const renderRoute = renderer.renderRoute.bind(renderer);
 
-  return require(path.resolve(driver)).default({}, invalidators);
+  renderer.renderRoute = (route, context) => {
+    const render = () => renderRoute(route, context)
+    return renderFn(route, context, render)
+  };
 }
 
-function cacheModule (options) {
-  Logger.info('Installed cache');
+/**
+ * Loads driver using path provided in the configuration.
+ */
+function createDriver (driver) {
+  return Array.isArray(driver)
+    ? requirePackage(driver[0])(driver[1])
+    : requirePackage(driver)();
+}
+
+/**
+ * Main VSF cache module.
+ */
+export default function cacheModule (options) {
+  Logger.info('Installed Vue Storefront Cache plugin');
 
   // This part must be before the condition below
-  const resolvedDriver = createDriver(options)
   this.addPlugin({
-    src: path.resolve(__dirname, './plugin.js'),
+    src: resolve(__dirname, './plugin.js'),
     mode: 'server',
-    options: {
-      ...options.server,
-      driver: resolvedDriver
-    }
+    options
   });
 
   if (!this.nuxt || !this.nuxt.renderer) {
-    return
+    return;
   }
-  const createRenderer = rendererFactory.bind(this)
-
-  if (options.server.invalidateEndpoint) {
-    const { middleware, extend } = createMiddleware({});
 
-    extend((app) => {
-      app.get(options.server.invalidateEndpoint, async (req, res) => {
-        await resolvedDriver.invalidate({ req, res });
+  // Create cache driver
+  const driver = createDriver(options.driver)
 
-        res.send()
-      })
-    })
+  // Create invalidation endpoint if necessary
+  createInvalidationEndpoint.call(this, driver, options.server);
 
-    this.addServerMiddleware(middleware);
-  }
-
-  createRenderer(async (route, context, render) => {
+  // Create renderer
+  createRenderer.call(this, async (route, context, render) => {
     const getTags = () => {
-      if (context.req.$vsfCache && context.req.$vsfCache.tagsSet) {
-        return Array.from(context.req.$vsfCache.tagsSet)
-      }
-
-      return []
+      const tags = context.req.$vsfCache && context.req.$vsfCache.tagsSet
+        ? Array.from(context.req.$vsfCache.tagsSet)
+        : [];
+      
+      return tags.map(({ prefix, value }) => `${prefix}${value}`);
     }
 
     try {
-      return await resolvedDriver.invoke({ route, context, getTags, render });
+      return await driver.invoke({
+        route,
+        context,
+        render,
+        getTags
+      });
     } catch (err) {
-      Logger.error('Your cache driver thrown an error!')
-      Logger.error('Server is going to render fresh page (cacheless)')
-      Logger.error(err)
+      Logger.error('Cache driver thrown an error when fetching cache! Server will render fresh page.');
+      Logger.error(err);
       return render();
```

Comment:
I imagine `render` is also async method. So I suggest keep the convention and either use `return await` or just `return` for both ones

Reply:
`await` before `driver.invoke` is required to catch errors. However, it's not required before `render`, because we can assume that it will work, and Promise returned from `async` function will still be awaited, so it doesn't give any benefit (WebStorm even show suggestions to remove `await`s from such places).